### ISSUE TITLE
Table: the row selection checkboxes are inaccessible for screen readers

### ISSUE THREAD

**Issue by Kiarokh** [Lundalogik/lime-elements#3636]
## Current behavior
The `limel-checkbox` component has accessibility support, as long as the consumer provides a `label` for it. Without a label, the checkbox becomes unidentifiable.

This becomes a problem in components like `limel-table` where there are unlabeled checkboxes that can be used to select "all rows" or individual rows. 
<img width="412" height="291" alt="Image" src="https://github.com/user-attachments/assets/252c66c5-8044-4b24-a9c4-0498a94c5ea6" />

## Expected behavior
Most probably due to lack of space, developers had chosen not to include a label for these checkboxes, or perhaps they had just missed this accessibility issue.

## Implementation
The `<label>` element needs to be rendered in the DOM for the checkbox to be interactive and clickable (suing pointing devices). This is the root cause is that we are visually hiding the native checkbox element in our `limel-checkbox` due to our styling needs, and instead add a non-clickable visual checkbox with custom styles and animations. Therefore, the only clickable part of the native HTML checkbox which is left is the `<label>` element. Without it, there's nothing for users to click on. 

In short:

1. The actual `<input>` element is hidden via CSS
2. The `<div class="box"` element has `pointer-events: none;`
3. Only the `<label>` element would then be clickable to trigger the checkbox

### How to fix this?
1. We need a way to keep the label rendered, but visually invisible; and make its physical size as big as the checkbox itself.
2. We need proper labels on checkboxes that are in `limel-table`

>[!important]
> #### Labels shouldn't toggle with the state
>1. Labels describe the control, not its current state.
>The label should describe what the control is or what it controls, not change depending on its state.
>
>2. Changing labels dynamically can confuse screen reader users.
>It’s better to provide consistent labels and convey changes through state (like checked/unchecked), rather than changing the label to a different meaning.
>3. Checkboxes are inherently binary controls.
>Their state is clear via checked/not checked. A checkbox labeled "Select all rows" that is checked implies all rows are selected; unchecking it implies deselection.
